src/infrastructure/jira/jira-service.ts

```diff
@@ -47,25 +51,32 @@ class JiraService {
 			connectInstallation,
 		);
 
-		if (response.rejectedEntities.length) {
-			const { key, errors } = response.rejectedEntities[0];
-			throw JiraServiceSubmitDesignError.designRejected(key.designId, errors);
-		}
+		this.throwIfSubmitDesignResponseHasErrors(response);
+	};
 
-		// TODO: Confirm whether we need to consider the use case below as a failure and throw or just leave a warning.
-		if (response.unknownIssueKeys?.length) {
-			throw JiraServiceSubmitDesignError.unknownIssueKeys(
-				response.unknownIssueKeys,
-			);
-		}
+	updateDesigns = async (
```

Comment:
**suggestion:** Consider renaming the method to `submitDesigns` for the following reasons:

- Conceptually, this method does "upsert" but not the "update" operation.
- It will be consistent with the existing `submitDesign` (which does the same thing but for 1 design).

Reply:
Also, consider refactor the existing `submitDesign` to delegate all the work to `submitDesigns` :)

Reply:
Great suggestion, I've renamed the method to `submitDesigns` and it now takes an array of `SubmitDesignParams`. I've also updated `submitDesign` to just call out to the new `submitDesigns` method.